build.sc

```diff
@@ -25,18 +27,18 @@ object InternalReleaseModule extends Module {
     * env variables to publish to sonatype for you.
     */
   def publishAll(ev: Evaluator): Command[Unit] = {
-    ReleaseModule.publishAll(ev)
+    release.publishAll(ev)
   }
 
   import Discover._
-  lazy val millDiscover: mill.define.Discover[this.type] =
+  lazy val millDiscover: mill.define.Discover =
     mill.define.Discover[this.type]
 }
 
 trait BaseModule extends Module with HeaderModule {
   def millSourcePath: os.Path = {
-    val originalRelativePath = super.millSourcePath.relativeTo(os.pwd)
-    os.pwd / "modules" / originalRelativePath
+    val originalPath = super.millSourcePath
+    (originalPath / os.up) / "modules" / originalPath.last
```

Comment:
Another option would be to do:

```scala
def millSourcePath: os.Path = {
  val originalPath =
    super.millSourcePath.relativeTo(mill.api.WorkspaceRoot.workspaceRoot)
  mill.api.WorkspaceRoot.workspaceRoot / "modules" / originalPath
}
```

We can't just use `Task.workspace` because `millSourcePath` is not a task.